GAE/src/org/waterforpeople/mapping/app/web/SurveyInstanceServlet.java

```diff
@@ -85,14 +90,21 @@ protected RestResponse handleRequest(RestRequest req) throws Exception {
         if (GET_INSTANCE_DATA_ACTION.equals(siReq.getAction())) {
             return retrieveInstanceData(siReq.surveyInstanceId);
         } else  if (GET_FORM_INSTANCES_ACTION.equals(siReq.getAction())) {
-            FormInstanceUtil formInstanceUtil = new FormInstanceUtil();
-            List<SurveyInstance> formInstances = formInstanceUtil.getFormInstances(siReq.getAndroidId(), siReq.getDataPointIdentifier(), LIMIT_FORM_INSTANCES_30, siReq.getCursor());
-
             FormInstanceResponse response = new FormInstanceResponse();
-            response.setSurveyInstances(formInstanceUtil.getFormInstancesDtoList(formInstances));
-            response.setCursor(BaseDAO.getCursor(formInstances));
-            response.setResultCount(formInstances.size());
+            FormInstanceUtil formInstanceUtil = new FormInstanceUtil();
 
+            try {
+                List<SurveyInstance> formInstances = formInstanceUtil.getFormInstances(siReq.getAndroidId(), siReq.getDataPointIdentifier(), LIMIT_FORM_INSTANCES_30, siReq.getCursor());
+                response.setSurveyInstances(formInstanceUtil.getFormInstancesDtoList(formInstances));
+                response.setCursor(BaseDAO.getCursor(formInstances));
+                response.setResultCount(formInstances.size());
+
+                return response;
+            } catch (Exception e) {
+                log.warning("Exception accessing endpoint: " + e.getMessage());
+                response.setCode(String.valueOf(HttpServletResponse.SC_NOT_FOUND));
```

Comment:
not sure if every exception is a not found exception 🤔 ... maybe we need a couple of specific exceptions to know which kind was thrown (eg: DeviceNotFoundException) from the formInstanceUtil

Reply:
then you enter in the *longish conversation* of checked vs unchecked exceptions. https://www.baeldung.com/java-checked-unchecked-exceptions